pkg/components/azure-arc/image/script.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Type of action: register or remove the Lokomotive cluster from Azure Arc.
+action=$1
+
+ca="$(cat /var/run/secrets/kubernetes.io/serviceaccount/ca.crt | base64 | tr -d "\n")"
+namespace="$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)"
+token="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token | tr -d "\n")"
+server="https://kubernetes.default.svc"
+
+echo "
+apiVersion: v1
+kind: Config
+clusters:
+- name: default-cluster
+  cluster:
+    certificate-authority-data: ${ca}
+    server: ${server}
+contexts:
+- name: default-context
+  context:
+    cluster: default-cluster
+    namespace: ${namespace}
+    user: default-user
+current-context: default-context
+users:
+- name: default-user
+  user:
+    token: ${token}
+" > sa.kubeconfig
+
+function azure_login() {
+  az login --service-principal -u ${AZURE_APPLICATION_CLIENT_ID} -p ${AZURE_APPLICATION_PASSWORD} --tenant ${AZURE_TENANT_ID}
+}
+
+function perform_action() {
+  case ${action} in
+    register)
+      # Connect the cluster to Azure Arc
+      az connectedk8s connect --kube-config sa.kubeconfig --name ${CONNECTED_CLUSTER_NAME} --resource-group ${AZURE_RESOURCE_GROUP}
```

Comment:
Is this commend idempotent?

Reply:
When the component is successfully installed (i.e job completed successfully), no matter how many times the `lokoctl component apply azure-arc-onboarding` is executed, it has no effect. i.e Job doesn't run again.

same with component delete.